lib/bugsnag/api/client/errors.rb

```diff
@@ -42,7 +42,15 @@ def update_errors(project_id, ids, operation, options = {})
           when String
             patch "projects/#{project_id}/errors/#{ids}", options.merge({:operation => operation})
           when Array
-            patch "projects/#{project_id}/errors", options.merge({:operation => operation, :query => {:error_ids => ids.join(' ')}})
+            defaults = {:operation => operation, :query => {:error_ids => ids.join(' ')}}
+            merged_opts = defaults.merge(options) do |key, oldVal, newVal|
+              if oldVal.is_a?(Hash) && newVal.is_a?(Hash)
+                oldVal.merge(newVal)
+              else
+                newVal
+              end
+            end
```

Comment:
Since this procedure is in three different places and across different files, it would be better to have it in a separate helpers file.

```diff
- merged_opts = defaults.merge(options) do |key, oldVal, newVal|
-   ...
- end
+ merged_opts = deep_merge(defaults, options)
```